File: src/main/java/com/matag/cards/properties/PowerToughness.java
package com.matag.cards.properties;

import lombok.Data;

import static java.lang.Integer.parseInt;

@Data
public class PowerToughness {
  private final int power;
  private final int toughness;

  public PowerToughness(int power, int toughness) {
    this.power = power;
    this.toughness = toughness;
  }

  public static PowerToughness powerToughness(String powerToughnessString) {
    var powerToughness = powerToughnessString.split("/");
    return new PowerToughness(parseInt(powerToughness[0]), parseInt(powerToughness[1]));
  }
}

File: src/main/kotlin/com/matag/cards/properties/PowerToughness.kt
package com.matag.cards.properties

data class PowerToughness(
    val power: Int,
    val toughness: Int
) {
    companion object {
        fun powerToughness(powerToughnessString: String): PowerToughness {
            val powerToughness: Array<String?> =
                powerToughnessString.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            return PowerToughness(powerToughness[0]!!.toInt(), powerToughness[1]!!.toInt())
        }
    }
}
